### ISSUE TITLE
[Bug]: Division by zero into lizmapTiler

### ISSUE THREAD

**Issue by laurentj** [3liz/lizmap-web-client#2438]
### What is the bug?

In log of a lizmap, there are tons of errors like this :

```
 warning  Division by zero
  lizmap/modules/lizmap/classes/lizmapTiler.class.php    line 243 and 244
index.php/lizmap/service/?repository=XXXXX&project=YYYYY&SERVICE=WMTS&REQUEST=GetCapabilities&VERSION=1.0.0
 array (   'repository' => 'XXXXX',   'project' => 'YYYYY',   'SERVICE' => 'WMTS',   'REQUEST' => 'GetCapabilities',   'VERSION' => '1.0.0',   'module' => 'lizmap',   'action' => 'service:index', )

        0       lizmapTiler::getTileMatrixSetList()     lizmap/modules/lizmap/classes/lizmapTiler.class.php : 93
        1       lizmapTiler::getTileCapabilities()      lizmap/modules/lizmap/classes/lizmapWMTSRequest.class.php : 36
        2       lizmapWMTSRequest->process_getcapabilities()   lizmap/modules/lizmap/classes/lizmapOGCRequest.class.php : 198
        3       lizmapOGCRequest->process()     lizmap/modules/lizmap/controllers/service.classic.php : 356
        4       serviceCtrl->GetCapabilities() lizmap/modules/lizmap/controllers/service.classic.php : 100
        5       serviceCtrl->index()   lib/jelix/core/jCoordinator.class.php : 225
        6       jCoordinator->process() index.php : 23
```

### Steps to reproduce the issue

unknown

### Lizmap version

3.4.4

### QGIS desktop version

3.16

### QGIS server version

3.16

### Operating system

?

### Browsers

Firefox

### Browsers version

?

**Comment by laurentj** [comment]
I take it.

Because of some issues about the cache (which I fixed in a first patch), the option mapScales  may be empty. The problem is that all calculus are based on a non empty mapScales. It it is empty in the cfg file, we have some "division by zero" errors. I will propose a patch to have a default mapScales.

But can we set a default mapScales?

**Comment by Gustry** [comment]
It's strange that you have empty `mapScales`.
Lizmap plugin can't save the CFG if the lizmap project mapscales is empty:

![Screenshot from 2021-09-03 08-03-56](https://user-images.githubusercontent.com/1609292/131958197-f51efcd8-a0b8-4fc0-9dea-c99b8081ffdb.png)

Are you referring to cfg file -> `options` -> `mapScales` ? or is-it another source ?

**Comment by mdouchin** [comment]
No @Gustry this property could have been emptied by LWC due to a bug in the cache management. The plugin is not concerned in this case I think

**Comment by laurentj** [comment]
No, it seems the plugin is not concerned, the cfg file I used has good values. Anyway, the code in lizmap should be more robust, and should check the validity of values to avoid bugs or such errors like "division by zero". The cfg file can be modified by hand, or lizmap could have a bug (like in our case).
